Presentation.Web/Controllers/API/V2/Internal/Notifications/Mapping/NotificationResponseMapper.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Core.Abstractions.Types;
+using Core.ApplicationServices.Model.Notification;
+using Core.DomainModel.Advice;
+using Core.DomainModel.GDPR;
+using Core.DomainModel.ItContract;
+using Core.DomainModel.ItSystemUsage;
+using Core.DomainModel.Shared;
+using Core.DomainServices.Generic;
+using Presentation.Web.Controllers.API.V2.Common.Mapping;
+using Presentation.Web.Models.API.V2.Internal.Request.Notifications;
+using Presentation.Web.Models.API.V2.Internal.Response.Notifications;
+using Presentation.Web.Models.API.V2.Response.Generic.Identity;
+
+namespace Presentation.Web.Controllers.API.V2.Internal.Notifications.Mapping
+{
+    public class NotificationResponseMapper : INotificationResponseMapper
+    {
+        private readonly IEntityIdentityResolver _identityResolver;
+
+        public NotificationResponseMapper(IEntityIdentityResolver identityResolver)
+        {
+            _identityResolver = identityResolver;
+        }
+
+        public Result<NotificationResponseDTO, OperationError> MapNotificationResponseDTO(Advice notification)
+        {
+            var type = notification.Type;
+            var ownerResourceUuidResult = ResolveRelationId(notification.RelationId.GetValueOrDefault(), type);
```

Comment:
it's odd that it's the responsibility of the data mapper to verify the correctness of the model right?